doc/developers/advanced_installation.rst

```diff
@@ -7,58 +7,102 @@ Advanced installation instructions
+
   * Install the version of scikit-learn provided by your
     :ref:`operating system or Python distribution <install_by_distribution>`.
-    This is the quickest option for those who have operating systems that
-    distribute scikit-learn.
-
-  * :ref:`Install an official release <install_official_release>`. This
-    is the best approach for users who want a stable version number
-    and aren't concerned about running a slightly older version of
-    scikit-learn.
+    This is a quick option for those who have operating systems
+    that distribute scikit-learn. It might not provide the latest release version.
 
-  * :ref:`Install the latest development version
+  * :ref:`Building the package from source
     <install_bleeding_edge>`. This is best for users who want the
     latest-and-greatest features and aren't afraid of running
-    brand-new code.
+    brand-new code. This document describes how to build from source 
 
 .. note::
 
     If you wish to contribute to the project, you need to
     :ref:`install the latest development version<install_bleeding_edge>`.
 
+.. _install_bleeding_edge:
 
-.. _install_official_release:
-
-Installing an official release
-==============================
+Building from source
+=====================
 
 Scikit-learn requires:
 
 - Python (>= 2.7 or >= 3.4),
 - NumPy (>= 1.8.2),
 - SciPy (>= 0.13.3).
 
+Building Scikit-learn also requires
 
-Mac OSX
--------
+- Cython >=0.23 
 
-Scikit-learn and its dependencies are all available as wheel packages for OSX::
+Running tests requires
 
-    pip install -U numpy scipy scikit-learn
+- pytest
 
+.. _git_repo:
 
-Linux
------
+Retrieving the latest code
+--------------------------
+
+We use `Git <https://git-scm.com/>`_ for version control and
+`GitHub <https://github.com/>`_ for hosting our main repository.
+
+You can check out the latest sources with the command::
+
+    git clone git://github.com/scikit-learn/scikit-learn.git
 
-At this time scikit-learn does not provide official binary packages for Linux
-so you have to build from source if you want the latest version.
-If you don't need the newest version, consider using your package manager to
-install scikit-learn. It is usually the easiest way, but might not provide the
-newest version.
+or if you have write privileges::
+
+    git clone git@github.com:scikit-learn/scikit-learn.git
+
+If you want to build a stable version, you can ``git checkout <VERSION>`` to get the code for that particular version, or download an zip archive of the version from github.
+
+Building from source
+--------------------
+If you have all the build requirements installed (see below for details), you can build and install the package in the following way.
+
+If you run the development version, it is cumbersome to reinstall the
+package each time you update the sources. It is thus preferred that
+you add the scikit-learn directory to your ``PYTHONPATH`` and build the
```

Comment:
I'm not sure about suggesting to modify `PYTHONPATH` as the preferred way to install from source. 

I think `pip install --editable .` + suggesting to use either the `--user` flag or run it in a virtualenv/conda virtualenv (for contributors) might be more standard.